src/node_serdes.cc

```diff
@@ -0,0 +1,457 @@
+}
+
+void SerializerContext::WriteDouble(const FunctionCallbackInfo<Value>& args) {
+  SerializerContext* ctx;
+  ASSIGN_OR_RETURN_UNWRAP(&ctx, args.Holder());
+
+  ctx->serializer_.WriteDouble(args[0]->NumberValue());
+}
+
+void SerializerContext::WriteRawBytes(const FunctionCallbackInfo<Value>& args) {
+  SerializerContext* ctx;
+  ASSIGN_OR_RETURN_UNWRAP(&ctx, args.Holder());
+
+  if (!args[0]->IsUint8Array()) {
+    return ctx->env()->ThrowTypeError("source must be a Uint8Array");
+  }
+
+  ctx->serializer_.WriteRawBytes(Buffer::Data(args[0]),
+                                 Buffer::Length(args[0]));
+}
+
+DeserializerContext::DeserializerContext(Environment* env,
+                                         Local<Object> wrap,
+                                         Local<Value> buffer,
+                                         Local<Value> readHostObject)
+  : BaseObject(env, wrap),
+    data_(reinterpret_cast<const uint8_t*>(Buffer::Data(buffer))),
+    length_(Buffer::Length(buffer)),
+    deserializer_(env->isolate(), data_, length_, this) {
+  object()->Set(env->context(), env->buffer_string(), buffer);
+  object()->Set(env->context(), env->read_host_object_string(), readHostObject);
+  deserializer_.SetSupportsLegacyWireFormat(true);
+
+  MakeWeak<DeserializerContext>(this);
+}
+
+MaybeLocal<Object> DeserializerContext::ReadHostObject(Isolate* isolate) {
+  Local<Value> readHostObject =
+      object()->Get(env()->context(),
+                    env()->read_host_object_string()).ToLocalChecked();
+
+  if (!readHostObject->IsFunction()) {
+    return ValueDeserializer::Delegate::ReadHostObject(isolate);
+  }
+
+  MaybeLocal<Value> ret =
+      readHostObject.As<Function>()->Call(env()->context(),
+                                          object(),
+                                          0,
+                                          nullptr);
+
+  if (ret.IsEmpty())
+    return MaybeLocal<Object>();
+
+  Local<Value> retValue = ret.ToLocalChecked();
+  if (!retValue->IsObject()) {
+    env()->ThrowTypeError("readHostObject must return an object");
+    return MaybeLocal<Object>();
+  }
+
+  return retValue.As<Object>();
+}
+
+void DeserializerContext::New(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsUint8Array()) {
+    return env->ThrowTypeError("buffer must be a Uint8Array");
+  }
+
+  if (!args[1]->IsFunction() && !args[1]->IsUndefined()) {
+    return env->ThrowTypeError("readHostObject must be a "
+                               "function or undefined");
+  }
+
+  new DeserializerContext(env, args.This(), args[0], args[1]);
+}
+
+void DeserializerContext::ReadHeader(const FunctionCallbackInfo<Value>& args) {
+  DeserializerContext* ctx;
+  ASSIGN_OR_RETURN_UNWRAP(&ctx, args.Holder());
+
+  Maybe<bool> ret = ctx->deserializer_.ReadHeader(ctx->env()->context());
+
+  if (ret.IsJust()) args.GetReturnValue().Set(ret.FromJust());
```

Comment:
Consistency: you just `.FromMaybe(false)` without an `if` elsewhere.  I think I like this approach better, though; it's Obviously Correct, even if it's a little more verbose.